watson_developer_cloud/watson_service.py

```diff
@@ -197,12 +231,11 @@ def __init__(self, vcap_services_name, url, username=None, password=None,
             self.default_headers = {'x-watson-learning-opt-out': 'true'}
 
         if api_key is not None:
-            if username is not None or password is not None:
-                raise ValueError(
-                    'Cannot set api_key and username and password together')
             self.set_api_key(api_key)
-        else:
+        elif username is not None and password is not None:
             self.set_username_and_password(username, password)
+        else:
```

Comment:
Should the token manager be the default? Is there an option to use unauthenticated?

Reply:
Got your point, instead if defaulting, it should do a check. I am updating this. Incase none of the authentication is set, it would throw an error